xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -411,11 +410,17 @@ void CDVDSubtitlesLibass::ApplyStyle(const std::shared_ptr<struct style>& subSty
     style->Bold = isFontBold * -1;
     style->Italic = isFontItalic * -1;
 
+    //! @todo Libass has a problem with color transparencies when set to:
+    //! PrimaryColour/SecondaryColour/OutlineColour by causing a gap between border
+    //! and text color. As workaround the SecondaryColour must have no transparency
+    //! this will fix just use cases without transparencies, for a full fix will be
+    //! needed in future update libass library having the gap fix.
+
     // Set default subtitles color
     style->PrimaryColour = ConvColor(subStyle->fontColor, subStyle->fontOpacity);
     // Set secondary colour for karaoke
     // left part is filled with PrimaryColour, right one with SecondaryColour
-    style->SecondaryColour = ConvColor(subStyle->fontColor, COLOR_OPACITY_30);
+    style->SecondaryColour = ConvColor(COLOR::BLACK);
```

Comment:
Any reason for forcing black instead of reusing `subStyle->fontColor` without touching alpha (setting opacity)?

Reply:
because if you set the same PrimaryColour color also here you will broke the karaoke style
so cannot distinguish text that should be highligthed from the background

Reply:
Hum, can you please adapt the comment just above the comment to explain/reflect this?

Reply:
so rewrite "left part is filled with PrimaryColour, right one with SecondaryColour" ?
something like
"the PrimaryColour is used to highlight text, the SecondaryColour is used as the colour of the passive text to be highlighted"

Reply:
No, just write something like "//! @bug in libass - force black secondary color to avoid reusing the same primary color. This would cause a visible gap in 'karaoke-style' subs"

Reply:
ah ok i add it

Reply:
however this bug is not 'karaoke-style' dependent, so the bug appears with every PrimaryColour/SecondaryColour/OutlineColour property when color transparency is set

Reply:
yeap please make it generic. Also a note, `karaoke-style` for those closed caption types is also not the correct word, in the cc_decoder the correct term is roll_up.

Reply:
cc roll-up is different thing so this is cc roll-up: https://www.youtube.com/watch?v=F_q-RRXw-vY&t=13s
where the text shift from one line to other in up or down direction
for libass should match "scrollup/down" feature

instead karaoke style on libass is similar to this: https://www.youtube.com/watch?v=vm-LI6QtKIc
where the text is successively highlighted one word at a time as doc explain